### PR TITLE
[ONNX] Fix an assertion failure involving Slice

### PR DESCRIPTION
**Pull request by yan12125**
Before this change, exporting a model to ONNX involving Slice crashes at `axes[i]` in line 153 if C++ assertions are enabled:
```
/usr/include/c++/11.1.0/bits/stl_vector.h:1045: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = long int; _Alloc = std::allocator<long int>; std::vector<_Tp, _Alloc>::reference = long int&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__n < this->size()' failed.
```
The relevant check is https://github.com/gcc-mirror/gcc/blob/releases/gcc-11.1.0/libstdc++-v3/include/bits/stl_vector.h#L1045, which checks the vector index.

The issue can be reproduced by exporting Mask R-CNN or similar ones. For example,
```Python
import io
import torch
import torchvision as tv

model = tv.models.detection.maskrcnn_resnet50_fpn(pretrained=False)
x = [torch.rand(3, 300, 400), torch.rand(3, 500, 400)]
with io.BytesIO() as f:
    torch.onnx.export(model, x, f, opset_version=11)
```
(extracted from [onnxoptimizer tests](https://github.com/onnx/optimizer/blob/master/onnxoptimizer/test/optimizer_test.py))

Tested environment: Arch Linux x86_64 with pytorch and torchvisoin installed from [the official repo](https://github.com/archlinux/svntogit-community/blob/packages/python-pytorch/trunk/PKGBUILD) and [AUR](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=python-torchvision), respectively.

Fixes #ISSUE_NUMBER: seems no similar reports yet. Checked https://github.com/pytorch/pytorch/issues with some keyword combinations `Assertion failed Slice`, `Assertion failed ONNX`, `Assertion failed vector`

### CODE REVIEW

**Review by garymm — APPROVED** [review]
@yan12125
Thanks!
If you open a new PR against branch `onnx_ms_1`, then we can take care of merging this into master later.
Otherwise we'll have to ask a Meta employee to merge it which will probably take quite some time.

**Comment by yan12125** [comment]
Oops sorry. Apparently the xhange base feature in GitHub is not good here. Will create a new clean PR instead.
